pygsuite/drive/file.py

```diff
@@ -0,0 +1,231 @@
+        ".odt": "application/vnd.oasis.opendocument.text",
+        ".pdf": "application/pdf",
+        ".docx": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+        ".doc": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+        ".epub": "application/epub+zip",
+    },
+    GoogleDocFormat.SHEETS: {
+        ".xlsx": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        ".xls": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        ".ods": "application/x-vnd.oasis.opendocument.spreadsheet",
+        ".pdf": "application/pdf",
+        ".csv": "text/csv",
+        ".tsv": "text/tab-separated-values",
+    },
+    GoogleDocFormat.DRAWINGS: {
+        ".jpeg": "image/jpeg",
+        ".jpg": "image/jpeg",
+        ".png": "image/png",
+        ".svg": "image/svg+xml",
+        ".pdf": "application/pdf",
+    },
+    GoogleDocFormat.SLIDES: {
+        ".pptx": "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+        ".ppt": "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+        ".odp": "application/vnd.oasis.opendocument.presentation",
+        ".pdf": "application/pdf",
+        ".txt": "text/plain",
+    },
+    GoogleDocFormat.SCRIPTS: {
+        ".json": "application/vnd.google-apps.script+json",
+    },
+}
+
+
+class File:
+    """Base class for a Google Drive File"""
+
+    def __init__(self, id: str = None, client: Optional[Resource] = None):
+
+        self.id = parse_id(id) if id else None
+        self.client = client or Clients.drive_client
+
+        # cache the metadata
+        self._metadata = None
+
+    @classmethod
+    def create(
+        cls,
+        name: Optional[str] = None,
+        mimetype: Optional[str] = None,
+        body: Optional[dict] = None,
+        media_body: Optional[Union[BytesIO, MediaFileUpload, MediaIoBaseUpload]] = None,
+        client: Optional[Resource] = None,
+    ):
+        """Create a new file in Google Drive.
+
+        Args:
+            name (str): Name of the file.
+            mimetype (str): Specified type of the file to create.
+            body (dict): Request body, if not provided, one is created with `name` and `mimetype` params.
+            media_body (BytesIO, MediaFileUpload, MediaIoBaseUpload): Content for the file.
+            client (Resource): client connection to the Drive API.
+        """
+
+        # establish a client
+        client = client or Clients.drive_client
+
+        # create request body
+        body = {"name": name, "mimeType": mimetype} if not body else body
+
+        # handle media conversion for bytes-like objects
+        if isinstance(media_body, BytesIO):
+            # if a mimetype is not provided, find best match
+            if not mimetype:
+                logging.warning("No mimetype specified, attempting to determine one.")
+
+                import magic
```

Comment:
nifty -is this an extra dependency?

Reply:
It is! Need to think a little bit about how we want to deal with this because unfortunately the setup is a little different on Windows vs. other systems.

For instance, on Windows users would need to do:

```
pip install python-magic
pip install python-magic-bin
```

Whereas on Debian/Ubuntu they'd need:

```
sudo apt-get install libmagic1
```

And for OSX:

```
brew install libmagic # for homebrew
port install file # for macports
```

All of this described [here](https://github.com/ahupp/python-magic#installation). While I really like the convenience of being able to try and determine the MIME type for a user when none is specified, I don't _love_ the reliance on these DLL / other files. Kept it as a gated import here to postpone inevitably needing to make something more elegant 😬